src/utils/tar.zig

```diff
@@ -136,19 +139,25 @@ pub fn parallelUntarToFileSystem(
                 file_count += 1;
 
                 const contents = try allocator.alloc(u8, file_size);
-                _ = try reader.readAtLeast(contents, file_size);
+                const actual_contents_len = try reader.readAtLeast(contents, file_size);
+                if (contents.len != actual_contents_len) {
```

Comment:
same thing here -- just easier to understand if file_size is explicit in the comparison